CRM/Core/BAO/RecurringEntity.php

```diff
@@ -115,6 +115,14 @@ class CRM_Core_BAO_RecurringEntity extends CRM_Core_DAO_RecurringEntity {
   const MODE_NEXT_ALL_ENTITY = 2;
   const MODE_ALL_ENTITY_IN_SERIES = 3;
 
+  public static function getSubscribedEvents() {
```

Comment:
@totten my only misgiving on this PR is that it adds a few functions with stingy addition of return hints (and possibly type hints). I think return hints are definitely preferred practice these days & it's probably better to set your IDE to recommend them (the EA plugin does this).

I think it's worth just reviewing the functions added in this PR & adding return hints & type hints where possible

Reply:
Pushed up a few commits to add more type-hints. Some notes:

* For `getSubscribedEvents()`, the type-signature and semantics are dictated by `EventSubscriberInterface`. I suppose, technically, it is legal for an implementation of `getSubscribedEvents()` to specify a narrower return-type than the interface (e.g. `getSubscribedEvents(): array` is narrower than `getSubscribedEvents()`). Though it feels odd to deviate from the `interface`.
* There are some new/internal functions (e.g. `normalizeListenerMap()` and `mergeListenerMap()`). These would be OK with more hints. Added via 60a1dbc.
* For  Symfony-style listeners (`on_*($event)` functions), I think it would be legit to always return `void`, so added that via e435f17. Although this reminds me an a (*slightly unpleasant*) fact... that sometimes `hook_*($arg1, $arg2, ...)` will return data. 65bc654 fixes support for that edge-case.
* I wouldn't agree PHP-language type-hints are "definitely preferred". Changes like phpunit's retroactive type-hinting seem like gratuitous dephell, IMHO. But, yeah, if it's a new or obscure function, and if the signature is amenable to type-hinting in the current language, then :+1:.